kornia/geometry/camera/pinhole.py

```diff
@@ -298,12 +289,10 @@ def project_points(self, point_3d: torch.Tensor) -> torch.Tensor:
             >>> pinhole.project_points(X)
             tensor([[5.6088, 8.6827]])
         """
-        R = self.rotation_matrix
-        t = self.translation_vector
-        point_3d_cam = torch.matmul(R, point_3d.view(3, 1)) + t
-        return project_points(point_3d_cam.view(point_3d_cam.shape[0], 3), self.camera_matrix)
+        point_3d_cam = transform_points(self.extrinsics, point_3d)
+        return convert_points_from_homogeneous(transform_points(self.intrinsics, point_3d_cam))
```

Comment:
One small optimization here could be to precompute the projection matrix
```python 
P = self.intrinsics @ self.extrinsicd
x = transform _points(P, point _3d)
```

Reply:
OK, revised and pushed to PR